src/msspreadsheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,550 @@
+                error err = error(WORKSHEET_ERROR_CODE, message = "Error ocurred while opening the worksheet.");
+                return err;
+            }
+        } else {
+            error err = error(WORKSHEET_ERROR_CODE, message = "Error occurred while accessing the Microsoft Graph API.");
+            return err;
+        }
+    }
+
+    # Create a worksheet on this workbook.
+    # + worksheetName - name of the worksheet to be created
+    # + return - A Worksheet client object on success, else returns an error
+    public remote function createWorksheet(string worksheetName) returns @tainted (Worksheet|error) {
+        http:Request request = new;
+        json payload = {"name": worksheetName};
+        request.setJsonPayload(payload);
+        http:Response|error httpResponse = self.workbookClient->post("/v1.0/me/drive/root:" +
+                    self.properties.path + self.properties.workbookName + ".xlsx:/workbook/worksheets", request);
+        int position = -1;
+        string sheetId = "";
+
+        if (httpResponse is http:Response) {
+            if (httpResponse.statusCode == HTTP_STATUS_CREATED) {
+                json|error response = httpResponse.getJsonPayload();
+                if (response is map<json>) {
+                    json sheetIdItem = response["id"];
+                    sheetId = sheetIdItem.toString();
+                    json positionItem = response["position"];
+
+                    int|error res1 = ints:fromString(positionItem.toString());
+                    if (res1 is int) {
+                        position = res1;
+                    } else {
+                        error err = error(WORKSHEET_ERROR_CODE, message = "Error ocurred while " +
+                                                                            "getting the worksheet position.");
+                        return err;
+                    }
+
+                    Worksheet workSheet = new(self.microsoftGraphConfig, self.properties.path,
+                                            self.properties.workbookName, sheetId, worksheetName, position);
+                    return workSheet;
+                } else {
+                    error err = error(WORKSHEET_ERROR_CODE, message = "Error ocurred while creating the worksheet.");
+                    return err;
+                }
+            } else {
+                error err = error(WORKSHEET_ERROR_CODE, message = "Error ocurred while creating the worksheet.");
+                return err;
+            }
+        } else {
+            error err = error(WORKSHEET_ERROR_CODE, message = "Error occurred while accessing the Microsoft Graph API.");
+            return err;
+        }
+    }
+
+    # Remove a worksheet from this workbook.
+    # + worksheetName - name of the worksheet to be removed
+    # + return - boolean true on success, else returns an error
+    public remote function removeWorksheet(string worksheetName) returns @tainted (boolean|error) {
```

Comment:
This function only returns `true` or `error` right? We usually model such functions to have an `error?` return type.

If the action fails we return an `error`. If it is successful we don't return anything (or return `()`).

```suggestion
    public remote function removeWorksheet(string worksheetName) returns @tainted error? {
```

This makes handling the result easier.

Reply:
Yes the function returned true for successful operation or error.
Ok, I updated the code based on the comment.